frame/data-feed/src/benchmarking.rs

```diff
@@ -0,0 +1,176 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2020 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! DataFeed pallet benchmarking.
+
+#![cfg(feature = "runtime-benchmarks")]
+
+use super::*;
+
+use frame_benchmarking::{benchmarks, whitelisted_caller};
+use frame_support::parameter_types;
+use frame_system::RawOrigin;
+
+use crate::Module as DataFeed;
+
+parameter_types! {
+	pub storage StorageArgument: DataType = DataType::U128(0);
+}
+
+fn add_tmp_provider<T: Trait>(provider: T::AccountId) {
+	let provider_lookup: <T::Lookup as StaticLookup>::Source = T::Lookup::unlookup(provider);
+	let _ = DataFeed::<T>::add_provider(RawOrigin::Root.into(), provider_lookup.clone());
+}
+
+fn add_data_info<T: Trait>(key: StorageKey, provider: T::AccountId) {
+	add_tmp_provider::<T>(provider.clone());
+	let data_info = super::FeededDataInfo {
+		key_str: "USD".as_bytes().to_vec(),
+		number_type: NumberType::FixedU128,
+		operation: Operations::Average,
+		schedule: 2.into(),
+	};
+	let _ = DataFeed::<T>::register_storage_key(RawOrigin::Signed(provider).into(), key, data_info);
+}
+
+benchmarks! {
+	_ {	}
+
+	register_storage_key {
```

Comment:
For this to be weighted correctly, you need to fill the vec with some upper limit of items.

This is true with any Vec. So you should probably have some configuration that specifies the max number of items, then some logic in the pallet that only allows this thing to grow if we are less than the max, and then benchmark when we are at the max.

let me know if this makes sense.

Reply:
Actually I add that before, for the lenght of `StorageKey` and `Url`. But when I execute benchmarks, the generate file show `WARNING! Some components were not used: ["n"]`, and the execution result is different every time. ( I find same case in substrate in other pallet, like system.remarks() and others.). Those case just simply remove the parameter in  WeightInfo, so that I remove them all in this datafeed module. 
Ok I would recover it and try again

Reply:
I add it